src/cdomains/valueDomain.ml

```diff
@@ -128,12 +131,13 @@ struct
     | `Array x -> CArrays.is_bot x
     | `Blob x -> Blobs.is_bot x
     | `Thread x -> Threads.is_bot x
+    | `Mutex -> false
```

Comment:
Should the value in a singleton lattice not both be considered `bottom` and `top` at the same time? Or is this `false` needed for some specific reason?

Reply:
I think I initially worried that having `is_bot_value Mutex = true` would make some states dead because one variable has a bottom value. I changed both to `true` now and tests indicate no issue with that, so it makes more sense indeed.